client/src/main/java/org/eclipse/hono/client/impl/DelegatedCommandSenderImpl.java

```diff
@@ -201,14 +197,10 @@ static String getTargetAddress(final String tenantId, final String deviceId) {
                 Objects.requireNonNull(deviceId));
     }
 
-    private static Message createDelegatedCommandMessage(final Message originalMessage, final String targetAddress,
-            final String replyToAddress) {
-        Objects.requireNonNull(targetAddress);
+    private static Message createDelegatedCommandMessage(final Message originalMessage, final String replyToAddress) {
         Objects.requireNonNull(originalMessage);
         // copy original message
         final Message msg = MessageHelper.getShallowCopy(originalMessage);
-        // set target address
-        msg.setAddress(targetAddress);
```

Comment:
why is this no longer necessary?

Reply:
It was only necessary because we used an anonymous relay link here. But with #1474 this has been changed. We use a non-empty link address now and therefore we don't need to explicitly set the message address.

The address of the original command message is supposed to be preserved here. This is relevant for the case that the message is to be delegated to a gateway.

Reply:
> The address of the original command message is supposed to be preserved here. 

it will be copied by the *getShallowCopy* method?

Reply:
yes (along with the other message properties).